### PR TITLE
LF-4165: Create animal creation summary container

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- Create functions to format DB animals/batches to summary format
  (to run tests: `pnpm test src/tests/animalUtils.test.js`)
- Update AddAnimalsSummaryCard ([story](http://localhost:6006/?path=/story/components-addanimalssummarycard--default)) and IconSummary ([story](http://localhost:6006/?path=/docs/components-addanimalssummarycard-iconsummary--docs)) to modify sex format (The design was updated: [mockup](https://www.figma.com/design/eMffC0h3JPlWtzNttxJpil/Litefarm-Updated?node-id=12490-123003&m=dev)) 
- Implement `AddAnimalSummary` container

Jira link: https://lite-farm.atlassian.net/browse/LF-4165

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Until API calls are integrated,
1. comment out line 71-73 in `packages/webapp/src/containers/Animals/AddAnimals/index.tsx`
2. modify `formResultData` in `packages/webapp/src/containers/Animals/AddAnimals/AddAnimalSummary/index.tsx`:
```
formResultData.animals = <response of POST "/animals">
formResultData.batches = <response of POST "/animal_batches">
``` 
<details>
  <summary>Sample responses</summary>

  ``` 
  formResultData.animals = [
    {
      id: 8,
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      sex_id: 1,
      name: null,
    },
    {
      id: 3,
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      sex_id: 1,
      name: null,
    },
    {
      id: 7,
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      sex_id: 2,
      name: null,
    },
    {
      id: 2,
      default_type_id: 2,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      sex_id: null,
      name: null,
    },
    {
      id: 11,
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      sex_id: 1,
      name: null,
    },
    {
      id: 13,
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      sex_id: 1,
      name: null,
    },
    {
      id: 19,
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: 1,
      custom_breed_id: null,
      sex_id: 2,
      name: null,
    },
  ];
  formResultData.batches = [
    {
      id: 5,
      farm_id: 'xxxxx',
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      name: null,
      notes: null,
      count: 100,
      photo_url: null,
      animal_removal_reason_id: null,
      removal_explanation: null,
      removal_date: null,
      sex_detail: [],
      internal_identifier: 5,
      group_ids: [2],
    },
    {
      id: 4,
      farm_id: 'xxxxx',
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      name: null,
      notes: null,
      count: 100,
      photo_url: null,
      animal_removal_reason_id: null,
      removal_explanation: null,
      removal_date: null,
      sex_detail: [],
      internal_identifier: 4,
      group_ids: [1],
    },
    {
      id: 3,
      farm_id: 'xxxxx',
      default_type_id: 2,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      name: null,
      notes: null,
      count: 100,
      photo_url: null,
      animal_removal_reason_id: null,
      removal_explanation: null,
      removal_date: null,
      sex_detail: [],
      internal_identifier: 2,
      group_ids: [2, 3],
    },
    {
      id: 2,
      farm_id: 'xxxxx',
      default_type_id: 1,
      custom_type_id: null,
      default_breed_id: null,
      custom_breed_id: null,
      name: null,
      notes: null,
      count: 50,
      photo_url: null,
      animal_removal_reason_id: null,
      removal_explanation: null,
      removal_date: null,
      sex_detail: [],
      internal_identifier: 3,
      group_ids: [1, 3],
    },
  ];
  ```

</details>

- [x] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by SayakaOno** [comment]
Thank you Duncan! I thought the animal basics would be the only full width page! 😳
(but once this PR is merged to the integration, I think it will become full width, so I will need to add `max-width: 1024px`.)

Do you think all the pages in this flow should be full width??

**Comment by kathyavini** [comment]
@Duncan-Brain and @SayakaOno that is the correct part of Figma! Those summary screens at the bottom are the most recent ones Loïc made.

**Comment by SayakaOno** [comment]
White background...! Thank you both, I think merging the integration fixed it!

**Review by Duncan-Brain — DISMISSED** [review]

File: packages/webapp/src/containers/Animals/AddAnimals/utils.ts [hunk 1]
```diff
@@ -21,3 +38,95 @@ export const formatAnimalDetailsToDBStructure = (data: any) => {
...
+  sexes: AnimalSex[];
+}
+
+export const formatDBAnimalsToSummary = (data: Animal[], config: Config): AnimalSummary[] => {
+  const animalsPerTypeAndBreed = {} as { [key: string]: AnimalSummary };
+  const { defaultTypes, customTypes, defaultBreeds, customBreeds, sexes } = config;
+
+  data.forEach((animal) => {
+    const typeBreedkey = getTypeBreedKey(animal);
+    const sexMap = getSexMap(sexes, true);
```
[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/containers/Animals/AddAnimals/utils.ts:81]: Thank you so much for spotting that, I tried to make it more efficient but made it worse instead 🤮🤮🤮🤮🤮!!!

Initially, I created the map without translating the keys and then realized they needed to be translated, so I decided to make the function more flexible. But I can't remember how I ended up exporting it 😂

File: packages/webapp/src/tests/animalUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
+/*
```
[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/tests/animalUtils.test.js:1]: Thank you 😊 Yes, I wrote the tests while I was writing the functions.
We don't always have tasks that need tests, but they can be really handy sometimes! ❤️

**Review by kathyavini — APPROVED** [review]
Looks great!!! And functions perfectly merged with your other branch too 😉

**Review by Duncan-Brain — APPROVED** [review]

**Pull request merged**